tests/filecheck/backend/riscv/canonicalize.mlir

```diff
@@ -123,6 +123,9 @@ builtin.module {
   %xor_bitwise_zero_r0 = riscv.xor %c0, %c1 : (!riscv.reg, !riscv.reg) -> !riscv.reg<a0>
   "test.op"(%xor_bitwise_zero_r0) : (!riscv.reg<a0>) -> ()
 
+  %shift_left_zero_r0 = riscv.slli %i2, 0 : (!riscv.reg) -> !riscv.reg
```

Comment:
I'm not a riscv expert but would it be good to make the input and output different registers as in the other examples?

Reply:
Ah I remembered now why the other tests have the result register specified. The default value for the return type when creating a riscv op is not allocated, and we want to test that the result value being allocated is preserved. Could you please make a similar change here?